src/script/lua_api/l_mainmenu.cpp

```diff
@@ -643,17 +643,33 @@ int ModApiMainMenu::l_create_world(lua_State *L)
 	const char *name	= luaL_checkstring(L, 1);
 	int gameidx			= luaL_checkinteger(L,2) -1;
 
+	std::string new_dir = fs::SanitizeFilename(name);
 	std::string path = porting::path_user + DIR_DELIM
-			"worlds" + DIR_DELIM
-			+ name;
+			+ "worlds" + DIR_DELIM + new_dir;
+
+	if (fs::PathExists(path)) {
+		path = path + "_";
+		for (int i = 1; i <= 100; i++) {
+			std::ostringstream test_path;
+			test_path << path << i;
+			if (!fs::PathExists(test_path.str())) {
+				path = test_path.str();
+				break;
+			}
+			if (i == 100) {
+				lua_pushstring(L, "Failed to find free directrory name for world");
+				return 1;
+			}
```

Comment:
I don't like that the value `100` is copied here.  I'd prefer you at least use a preprocessor constant like the player file code uses.  (In fact, I'd you should use the exact same code since the issue you've fixed with invalid names also exist with player files)  They should both call a `find_safe_filename()` function.